container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,139 @@
+// It writes out both a config file and a manifest for the v2.2 image.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/bazelbuild/rules_docker/container/go/pkg/compat"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+var (
+	baseConfig         = flag.String("baseConfig", "", "The base image config.")
+	baseManifest       = flag.String("baseManifest", "", "The base image manifest.")
+	outputConfig       = flag.String("outputConfig", "", "The output image config file to generate.")
+	outputManifest     = flag.String("outputManifest", "", "The output manifest file to generate.")
+	creationTimeString = flag.String("creationTime", "", "The creation timestamp. Acceptable formats: Integer or floating point seconds since Unix Epoch, RFC, 3339 date/time.")
+	user               = flag.String("user", "", "The username to run the commands under.")
+	workdir            = flag.String("workdir", "", "Set the working directory of the layer.")
+	nullEntryPoint     = flag.String("nullEntryPoint", "False", "If True, Entrypoint will be set to null.")
+	nullCmd            = flag.String("nullCmd", "False", "If True, Cmd will be set to null.")
+	operatingSystem    = flag.String("operatingSystem", "linux", "Operating system to create docker image for, eg. linux.")
+	labelsArray        arrayFlags
+	ports              arrayFlags
+	volumes            arrayFlags
+	entrypointPrefix   arrayFlags
+	env                arrayFlags
+	command            arrayFlags
+	entrypoint         arrayFlags
+	layerDigest        arrayFlags
+	stampInfoFile      arrayFlags
+)
+
+const (
+	// defaultProcArch is the default architecture type based on legacy create_image_config.py.
+	defaultProcArch = "amd64"
+	// defaultTimeStamp is the unix epoch 0 time representation in 32 bits.
+	defaultTimestamp = "1970-01-01T00:00:00Z"
+)
+
+// arrayFlags can be used to store multiple flags the same name.
+// the resulting data parsed in will be an array data type.
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return fmt.Sprintf("%s", strings.Join(*i, ", "))
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+func main() {
+	flag.Var(&labelsArray, "labels", "Augment the Label of the previous layer.")
+	flag.Var(&ports, "ports", "Augment the ExposedPorts of the previous layer.")
+	flag.Var(&volumes, "volumes", "Augment the Volumes of the previous layer.")
+	flag.Var(&entrypointPrefix, "entrypointPrefix", "Prefix the Entrypoint with the specified arguments.")
+	flag.Var(&env, "env", "Augment the Env of the previous layer.")
+	flag.Var(&command, "command", "Override the Cmd of the previous layer.")
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layerDigest, "layer", "Layer sha256 hashes that make up this image. The order that these layers are specified matters.")
```

Comment:
I think these are files that contain the layerDigest (according to how it's being used in OverrideConfig), so I suggest naming the variable and the flag `layerDigestFile` instead and changing the description to say it's the file that contains the sha256 digest of the layer.

Also, I would explore the possibility of not having an `@` at the beginning of these names to avoid having to do what `extractValue` initially does to remove the `@` prefix.